.gitignore

```diff
@@ -37,5 +37,8 @@ jacoco.exec
 captures/
 __pycache__
 
+# Xcode cache for demo project
+DerivedData/
```

Comment:
Why it's inside project folder (instead of `~/Library/Developer/Xcode/DerivedData`)? Do we have it in config somewhere?

Reply:
It's a config of Xcode application itself. Very handy to ditch all caches during execution of regenerate_xcode_project.sh
<img width="830" alt="Screenshot 2023-07-19 at 11 08 10" src="https://github.com/JetBrains/compose-multiplatform-core/assets/4167681/c06000fc-1148-4662-a897-eabc33c821b3">

Reply:
I don't think that we need to add custom value from your local config to ignore

Reply:
There is no convinient way to ignore files in git on the local level.

If someone in the team works with some configuration/plugin - we can add it to .gitignore.

Reply:
`core.excludesFile` in a global .gitconfig might help though, I only tried `git update-index`, which is very unconvinient